src/quo2/components/counter/__tests__/counter_component_spec.cljs

```diff
@@ -0,0 +1,34 @@
+(ns quo2.components.counter.--tests--.counter-component-spec
```

Comment:
A heuristic I use to decide *what to test* in components is to think about input -> output, especially for pure functions or even in Javascript (components without state, hooks, etc).

The `counter` component is a pure function and it receives the following arguments, but not all of them are covered:

- `type`
- `override-text-color`
- `override-bg-color`
- `style`
- `accessibility-label`
- `value`

The tests should ideally cover the impact in the output of each input (in tandem or isolated, it doesn't matter much) because the **output only changes if the input changes** (pure function).

But I understand you're just trying to increase test coverage and give more examples, which is great :100: The other heuristic I saw you used in other tests is to test the behavior of event triggers, this is also a solid approach for stateful components!

Reply:
Yep, I 100% agree. ideally we have a test to cover each combination of various props.
I was trying to figure out how best to test pieces like `override-text-color` without too tightly binding to the implementation. Actually that's where I feel visual tests will work better as a testing mechanism. We can figure out an approach that best suits us though, as obviously visual tests will be much slower to run so perhaps better to do with tests such as these.

But yes, this p.r (and some follow ups) are just to create some broad coverage on components and have at the very minimum basic rendering covered.

Reply:
Thank you so much, incredibly valuable for us to eventually have a good "test pyramid".